test/js/bun/plugin/plugins.test.ts

```diff
@@ -476,7 +476,11 @@ describe("errors", () => {
         return new Response(result);
       },
     });
-    const { default: text } = await import(`http://${server.hostname}:${server.port}/hey.txt`);
+    const sleep = ms => new Promise<string>(res => setTimeout(() => res("timeout"), ms));
```

Comment:
This forces the test to take a minimum of 2.5 seconds

Reply:
<img width="598" alt="image" src="https://github.com/user-attachments/assets/a7dd98d1-f1ce-46d8-96d1-e5636f24fccd" />

Reply:
If you prefer we can use something like this:
```ts
export function timeout(timeout: number): CancellablePromise {
  const { promise, resolve, reject } = Promise.withResolvers<void>();

  const handle = setTimeout(() => {
    reject(new TimeoutError(timeout));
  });

  const cancellablePromise = Object.assign(promise, {
    cancel() {
      clearTimeout(handle);
      resolve();
    },
  });

  cancellablePromise.cancel = cancellablePromise.cancel.bind(cancellablePromise);
  return cancellablePromise;
}
```